.circleci/codecov.sh

```diff
@@ -0,0 +1,1733 @@
+#!/usr/bin/env bash
```

Comment:
We have a `./build-support` folder, let's put this script there instead of .circleci, unless there's a specific reason not to